mcs/class/System/Test/System.Net.Sockets/SocketTest.cs

```diff
@@ -3464,32 +3464,32 @@ public void ReceiveRemoteClosed ()
 		[Test]
 		public void UdpDoubleBind ()
 		{
-			Socket s = new Socket (AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
-			s.SetSocketOption (SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, 1);
-			
-			s.Bind (new IPEndPoint (IPAddress.Any, 12345));
-			
-			Socket ss = new Socket (AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
-			ss.SetSocketOption (SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, 1);
-			
-			ss.Bind (new IPEndPoint (IPAddress.Any, 12345));
+			using (Socket s = new Socket (AddressFamily.InterNetwork,
+						SocketType.Dgram, ProtocolType.Udp))
+			using (Socket ss = new Socket (AddressFamily.InterNetwork,
+						SocketType.Dgram, ProtocolType.Udp)) {
+				var supportsReuseAddress = true;
+				try {
+					s.SetSocketOption (SocketOptionLevel.Socket, SocketOptionName.ReuseAddress, true);
+				} catch (SocketException e) {
+					// Exception is thrown when ReuseAddress is not supported
```

Comment:
you should check here that e.NativeErrorCode == (int) SocketError.OperationNotSupported